### ISSUE TITLE
onRestError failures hidden

### ISSUE THREAD

**Issue by tuckerzp** [EasyDynamics/oscal-react-library#584]
Also, it looks like the failure above highlights an issue with the `onRestError` prop that we're using... somewhere we're failing to actually pass a function.

```
FAIL src/components/OSCALJsonEditor.test.js
  ● OSCALJsonEditor › should render as expected

    TypeError: props.onRestError is not a function



      at src/components/OSCALControlProse.js:605:23
      at src/components/oscal-utils/OSCALRestUtils.js:153:13

  ● OSCALJsonEditor › should render as expected

    TypeError: props.onRestError is not a function



      at src/components/OSCALControlProse.js:605:23
      at src/components/oscal-utils/OSCALRestUtils.js:153:13

  ● OSCALJsonEditor › should render as expected

    TypeError: props.onRestError is not a function



      at src/components/OSCALControlProse.js:605:23
      at src/components/oscal-utils/OSCALRestUtils.js:153:13

  ● OSCALJsonEditor › should render as expected

    TypeError: props.onRestError is not a function



      at src/components/OSCALControlProse.js:605:23
      at src/components/oscal-utils/OSCALRestUtils.js:153:13

  ● OSCALJsonEditor › should return JSON after clicking save

    connect ECONNREFUSED 127.0.0.1:80



  ● OSCALJsonEditor › should return JSON after clicking save

    connect ECONNREFUSED 127.0.0.1:80



  ● OSCALJsonEditor › should return JSON after clicking save

    connect ECONNREFUSED 127.0.0.1:80



  ● OSCALJsonEditor › should return JSON after clicking save

    connect ECONNREFUSED 127.0.0.1:80
```

This probably would have been prevented with stronger type checking. In any case, if it's trivial, we can fix it here. Otherwise, we should we probably open a separate issue.

_Originally posted by @kylelaker in https://github.com/EasyDynamics/oscal-react-library/issues/581#issuecomment-1224895904_

**Comment by brian-easyd** [comment]
Must specify React component property types before we can size or work this.

**Comment by brian-easyd** [comment]
@kylelaker do we have an issue for specifying React component property types that we can cite above?

**Comment by tuckerzp** [comment]
It looks like the logs we originally cited this from are gone. I agree with @kylelaker's original thought that progressively switching to `.tsx` and `.ts` will lead to this issue being fixed. 

If I had to guess, switching `OSCALJsonEditor.test.js` --> `OSCALJsonEditor.test.tsx` might be a good starting point. But it may not be worth the effort to actively try to replicate this bug if it just effects the tests and will naturally be fixed anyways. 

> do we have an issue for specifying React component property types that we can cite in the comment above?

I don't think we do but I also don't think adding one would provide a ton of value. So far, we have just been adding property type specifications when we switch files to `.tsx` or `.ts`. Rather than having one big issue that touches the entire code base, it has made more sense to just use the boy scout rule when we make changes in various files.
